cairo-libs/packages/protobuf/src/types/message.cairo

```diff
@@ -63,6 +69,25 @@ pub impl EncodeContextImpl of EncodeContextTrait {
             i += 1;
         }
     }
+
+    /// Performs the Protobuf encoding for an enum field.
+    fn encode_enum<T, +Drop<T>, +Into<T, u32>>(
```

Comment:
good idea of using `+Into<T, u32>>` for enums. 👍🏼

Reply:
I think, I didn't implement this way, because I wanted to keep the `encode_field` or `encode_repeated_field` as the top method to be called when implementing `ProtoMessage` on local struct. I don't mind having `encode_enum` and `encode_oneof` as long as, it is called inside `encode_field` and can be generalized for `encode_repeated_field` too.